fusion/fusion/fusion.h

```diff
@@ -0,0 +1,262 @@
+                   fusion::ShrPtrAlloc<MachineInfoType>(),
+               const FieldExtractorType field_extractor_alloc =
+                   fusion::ShrPtrAlloc<FieldExtractorType>()) :
+            fusiongroup_alloc_(fusiongroup_alloc),
+            machine_info_alloc_(machine_info_alloc),
+            fusionOpr(defaultFusionOpr)
+        {
+            if (fusiongroup_list.size() > 0)
+                initialize(fusiongroup_list);
+            else if (fusiongroupcfg_list.size() > 0)
+                initialize(fusiongroupcfg_list);
+            else if (txt_file_list.size() > 0)
+                initialize(txt_file_list);
+            context_.makeContext("fbase", fusiongroup_list);
+        }
+
+        //! \brief ctor from group list
+        Fusion(const FusionGroupListType & fusiongroup_list) :
+            Fusion(fusiongroup_list, {}, {}, FusionGroupTypeAlloc(),
+                   MachineInfoTypeAlloc(), FieldExtractorType())
+        {
+        }
+
+        //! \brief ctor from cfg group list
+        Fusion(const FusionGroupCfgListType & fusiongroupcfg_list) :
+            Fusion({}, fusiongroupcfg_list, {}, FusionGroupTypeAlloc(),
+                   MachineInfoTypeAlloc(), FieldExtractorType())
+        {
+        }
+
+        //! \brief ctor from txt file list
+        //!
+        //! This is for future support of the DSL or JSON.
+        //! The plan a constructor helper determines file type
+        Fusion(const FileNameListType & txt_file_list) :
+            Fusion({}, {}, txt_file_list, FusionGroupTypeAlloc(),
+                   MachineInfoTypeAlloc(), FieldExtractorType())
+        {
+        }
+
+        //! \brief bare ctor
+        Fusion() :
+            Fusion({}, {}, {}, FusionGroupTypeAlloc(),
+                   MachineInfoTypeAlloc(), FieldExtractorType())
+        {
+        }
+
+        //! \brief initialize state from a group list
+        void initialize(const FusionGroupListType & fusiongroup_list)
+        {
+            for (auto grp : fusiongroup_list)
+            {
+                registerGroup(grp);
+            }
+        }
+
+        //! \brief initialize from a text file list
+        //!
+        //! For simplicity assume all files are the same type
+        //! as first file. I do not see a need at the moment
+        //! for mixing types.
+        void initialize(const FileNameListType & txt_file_list)
+        {
+            if (txt_file_list.size() == 0)
+                return;
+            bool useJson = isJsonFile(txt_file_list[0]);
+            if (useJson)
+            {
+            }
+            else
+            {
+                qp.setInputFiles(txt_file_list);
+                int ret = qp.parse();
+                if (ret)
+                {
+                    // throw fusion::DslSyntaxError(qp.errMsg, qp.lineNo);
+                }
+            }
+        }
+
+        //! \brief future file type detector
+        bool isJsonFile(std::string fn) { return false; }
```

Comment:
method can be `const`.  probable want to `(void)` out `fn` or drop it.

Reply:
fixed